### DISCUSSION TITLE
[SO] Why RxJS `startWIth` does not emit when used with `lastValueFrom`?

### DISCUSSION THREAD

**Discussion by vajahath** [ReactiveX/rxjs#7329] [Q&A] [answered]
[SO Question](https://stackoverflow.com/questions/77013989/why-rxjs-startwith-does-not-emit-when-used-with-lastvaluefrom)

Short summary:

Why this thing does not resolve at all?
```ts
const obs = new Subject().asObservable().pipe(startWith('startWith'));
lastValueFrom(obs).then(console.log);
```

**Comment by demensky** [comment, marked as answer]
[`lastValueFrom`](https://rxjs.dev/api/index/function/lastValueFrom):

> Converts an observable to a promise by subscribing to the observable, **waiting for it to complete**…

In your case "complete" never happens.

Here is an example where `console.log` will be executed:

```ts
const sub = new Subject();
const obs = sub.asObservable().pipe(startWith('startWith'));
lastValueFrom(obs).then(console.log);

sub.complete();
```
